### ISSUE TITLE
deleting chunks slows down replication

### ISSUE THREAD

**Issue by onlyjob** [moosefs/moosefs#448]
I have some undergoal (endangered) chunks and they are being replicated to a certain chunkserver as expected.
Meanwhile I've deleted about a million files from MooseFS and, as expected, chunks deletion was active for about 4 hours.
Problem: during 4 hours of chunks deletion, replication rate dropped to 30% of the original speed and recovered only after deletion was over. Most of deleted chunks were sitting on chunkserver where undergoal chunks were replicated from (and no chunks were deleted from the chunkserver where undergoal chunks were replicating to). (Observed on 3.0.116).

So deleting chunks inhibited replication rate which is mildly problematic because from data safety prospective restoration of endangered chunks should have higher priority than removing already deleted data from trash.

I'm not sure why this is happening -- maybe because master processes only certain number of chunks per hour and deletion takes significant portion of queue. Or maybe because deletion slowed down chunkserver with the only copy of data to 30% of its performance. Either way it looks like there might be a room for improvement or optimisation.

**Comment by chogata** [comment]
Deleting is a very "heavy" operation. I don't mean deleting on MooseFS right now, but deleting in general, on local filesystems too. We observed that machines performing chunk deletions significantly drop in overall performance.
However, we cannot always assume that deleting is "much less important" than other operations on the filesystem. We might run into problems when we need that space or some other scenario that I cannot think about right now. In the past, we tried stronger prioritisation of some operations that we deemed "very important", and it always, ALWAYS ended up having the opposite effect at some point. We learned from our mistakes and no longer do that.
Of course, endangered replications ARE of a higher priority than non-endangered replications, which in turn are of a higher priority than deletions. However, lower priority never means complete stop to any single type of operations (unless the user explicitly blocks them in config by setting 0 limit). In your particular case simply lowering the deletion limit in config would help. If this is a common occurrence in your system and you can afford, space-wise, to keep chunks in undeleted state for a longer time, you could even make that setting permanent for you instance. But that's not something we dare make automatic, for the reason I mentioned above.

**Comment by onlyjob** [comment]
Thanks for your comments. You are right, I see no easy way to address the problem without causing side effects. Let's close this issue due to absence of actionable tasks.
